woocommerce/woocommerce
# [Experimental] Refact Rating filter block for the new Product Filters block structure

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR updates the **Rating Filter** block to work correctly with the new structure of the **Products Filter**.

### Disclaimer

This RP does not handle the Dropdown mode of the filter. I suggest to table it in a follow-up PR



Closes #51639



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Ensure the experimental blocks are enabled:
    - If you use the WooCommerce Beta Tester plugin, you can enable the experimental blocks in Tools > WCA Test Helper > Features. Toggle the `experimental-blocks` option.
    - If you build this PR locally, build the branch with `pnpm --filter='@woocommerce/plugin-woocommerce' build` command to have experimental blocks available.
1. Add the **Product Filters (Experimental)** block to the Product Catalog template.
2. Save and go to the shop page.
4. Select some filters, see filter blocks react, and update the filter options accordingly.
5. Interact with the **Rating** filter slider, see the filter updated accordingly.

https://github.com/user-attachments/assets/5d42339a-130d-4274-b9ad-7953943a7d1d



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
